src/realm/array.hpp

```diff
@@ -144,6 +144,11 @@ class Array : public Node, public ArrayParent {
     /// just the reference to the underlying memory. All elements will be
     /// initialized to the specified value.
     static MemRef create_array(Type, bool context_flag, size_t size, int_fast64_t value, Allocator&);
+    /// Construct a compressed integer array. Return the ref to the underlying memory.
+    /// All the elements will be initialized to the specif value, the calle will then need
+    /// to fill the arrat with the specific elements.
+    static MemRef create_flex_array(Type type, bool context_flag, size_t array_value_elements, size_t value_bit_width,
```

Comment:
I am not sure if this belongs to `Array` at this point, probably it is better if the most derived array implements the right logic, or we introduce a factory for these compressed arrays... something to evaluate...